### PR TITLE
Update Tokio to Rust 2018

### PR DESCRIPTION
**Pull request by carllerche**
This is the start to what will most likely be the long process of updating Tokio to async / await.

The first step is to update Tokio to use Rust 2018. This also bumps crate versions and sets all crates to `publish = false`.

### CODE REVIEW

File: tokio-signal/src/windows.rs [hunk 1]
```diff
@@ -158,7 +155,7 @@ fn global_init(handle: &Handle) -> io::Result<()> {
     let ready = reg.readiness.clone();
 
     let (tx, rx) = mpsc::unbounded();
-    let reg = try!(PollEvented::new_with_handle(reg, handle));
+    let reg = r#try!(PollEvented::new_with_handle(reg, handle));
```
Comment (ipetkov) [inline tokio-signal/src/windows.rs]:
```suggestion
    let reg = PollEvented::new_with_handle(reg, handle)?;
```

Comment (carllerche) [reply tokio-signal/src/windows.rs]: What was `cargo fix` thinking o_O

**Comment by carllerche** [comment]
I probably should update doc comments too.

**Comment by carllerche** [comment]
Ok... this should be good to go.

**Comment by CryZe** [comment]
I just tried the hyper example and it fails on: `thread 'tokio-runtime-worker-0' panicked at 'called `Result::unwrap()` on an `Err` value: Error(Inner(Error(User(Execute), "tokio::spawn failed (is a tokio runtime running this future?)")))', src\libcore\result.rs:999:5`

**Pull request merged**